### PR TITLE
Add ERC for wallet_getOwnedAssets

### PR DESCRIPTION
**Pull request by loredanacirstea**
Add ERC for `wallet_getOwnedAssets`

This is a proposal targeting wallet API standardization: a new JSON-RPC method for retrieving a selection of owned assets by an Ethereum address, with the user's permission.

### CODE REVIEW

**Comment by ligi** [comment]
Thanks for the initiative! I think the chainId should also be part of this EIP as users can own tokens on different chains. This might also help us on the path to ETH 2.0

**Comment by loredanacirstea** [comment]
@ligi , good idea. I added `chainId` in https://github.com/ethereum/EIPs/pull/2256/commits/b9701bbfe9b0dc689df2003da240da7f5fefa5c1

**Comment by danfinlay** [comment]
This proposal looks good, consider this my endorsement.

One addition that could be nice could be an optional `justification` field, where the dapp could provide a string of human-readable text explaining the intended purpose of the selected asset. It would be up to the wallet to distinguish this text from the rest of its UI, making it clear this was dapp-provided context.

This last reading also makes me think there is an opportunity for another method here:

Upon this latest review, I noticed a distinction between two types of "restricted methods".

1. Methods that may be approved in an ongoing way, per [web3 login permissions](https://ethereum-magicians.org/t/web3-login-permissions/3583)
2. Methods that require a prompt per interaction, like tx approval, or selecting a file for upload.

I think early on when I read this, I thought it was type 1, but now on closer review I notice it seems to be more type 2, since it includes constraints on the request itself, like `count` and `types`.

I think that's fine, but I now think we might want a second proposal of type 1 too, since for example you may want an exchange to always be aware of the tokens you are interested in, even if you add more to your list. The current proposal seems especially suited to per-selection interactions, like "select the item to put up for auction".

Under method 1, the method might be like this instead:

```javascript
wallet.requestPermissions({
  'wallet_selectOwnedAssets': {
    caveats: [
      { 
        type: 'eth_assetTypes',
        value: ['erc20', 'erc721']
      }
    ]
  }
})
```
Anyways, this would be a matter for another ERC, none of it is blocking, although `justification` might be nice to include.

File: EIPS/eip-2256.md [hunk 1]
```diff
@@ -0,0 +1,183 @@
+---
+eip: 2256
+title: wallet_getOwnedAssets JSON-RPC Method
+author: Loredana Cirstea (@loredanacirstea)
+discussions-to: https://ethereum-magicians.org/t/eip-2256-add-wallet-getownedassets-json-rpc-method/3600
+status: Draft
+type: Standards Track
+category: ERC
```
Comment (axic) [inline EIPS/eip-2256.md]: I think this belongs under Interface where the other RPCs are.

Comment (loredanacirstea) [reply EIPS/eip-2256.md]: done.

**Comment by noncash15** [comment]
> Add ERC for `wallet_getOwnedAssets`
> 
> This is a proposal targeting wallet API standardization: a new JSON-RPC method for retrieving a selection of owned assets by an Ethereum address, with the user's permission.
> 
> 

1c29cecb2916582fdd4ad7cf77acd7649fe5a0f6

File: EIPS/eip-2256.md (hunk 1 above)
Comment (noncash15) [reply EIPS/eip-2256.md]: EIPS/eip-2256.md

**Comment by noncash15** [comment]
#2256

**Pull request merged**